src/value/owned.rs

```diff
@@ -234,6 +293,26 @@ impl OwnedElement {
     }
 }
 
+impl PartialEq for OwnedElement {
+    fn eq(&self, other: &Self) -> bool {
+        match &self.ion_type() {
+            IonType::Null => self.is_null() == other.is_null(),
+            IonType::Boolean => self.as_bool() == other.as_bool(),
+            IonType::Decimal => self.as_decimal() == other.as_decimal(),
+            IonType::Integer => self.as_any_int() == other.as_any_int(),
+            IonType::Float => self.as_f64() == other.as_f64(),
+            IonType::Timestamp => self.as_timestamp() == other.as_timestamp(),
+            IonType::Clob | IonType::Blob => self.as_bytes() == other.as_bytes(),
+            IonType::String => self.as_str() == other.as_str(),
+            IonType::Symbol => self.as_sym() == other.as_sym(),
+            IonType::Struct => self.as_struct() == other.as_struct(),
+            IonType::SExpression | IonType::List => self.as_sequence() == other.as_sequence(),
+        }
+    }
+}
```

Comment:
Is it possible to lean on a `PartialEq` defined for `OwnedValue`?  Also don't you need to compare the annotations?  This needs its own explicit tests too.